localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,406 @@
+import logging
+import os
+import sqlite3
+from datetime import datetime, timezone
+from typing import Dict, List, Optional
+
+from localstack import config
+from localstack.aws.api.cloudwatch import MetricData, MetricDataQuery, ScanBy
+from localstack.utils.files import mkdir
+
+LOG = logging.getLogger(__name__)
+
+STAT_TO_SQLITE_FUNC = {
+    "Sum": "SUM(value)",
+    "Average": "AVG(value)",
+    "Minimum": "MIN(value)",
+    "Maximum": "MAX(value)",
+    "SampleCount": "COUNT(value)",
+}
+
+STAT_TO_SQLITE_COLUMNS = {
+    "Sum": "sum",
+    "Average": "sum, sample_count",
+    "Minimum": "min",
+    "Maximum": "max",
+    "SampleCount": "sample_count",
+}
+
+
+class CloudwatchDatabase:
+    DB_NAME = "metrics.db"
+    CLOUDWATCH_DATA_ROOT: str = os.path.join(config.dirs.data, "cloudwatch")
+    METRICS_DB: str = os.path.join(CLOUDWATCH_DATA_ROOT, DB_NAME)
+    TABLE_SINGLE_METRICS = "SINGLE_METRICS"
+    TABLE_AGGREGATED_METRICS = "AGGREGATED_METRICS"
+
+    def __init__(self):
+        if os.path.exists(self.METRICS_DB):
+            LOG.debug(f"database for metrics already exists ({self.METRICS_DB})")
+            return
+
+        mkdir(self.CLOUDWATCH_DATA_ROOT)
+        with sqlite3.connect(self.METRICS_DB, isolation_level="EXCLUSIVE") as conn:
+            cur = conn.cursor()
+            common_columns = """
+                "id"	                INTEGER,
```

Comment:
just making sure we want an integer as ID, since you usually don't perform arithmetic operations on them

Reply:
I think  Id as INTEGER is common practice in Relational DBs. But @steffyP can talk more about this.

Reply:
I think using the auto-increment feature of sqlite for the [INTEGER PRIMARY KEY](https://www.sqlite.org/datatypes.html) makes our life lot easier:

> INTEGER PRIMARY KEY columns can be used to implement the equivalent of AUTOINCREMENT. If you try to insert a NULL into an INTEGER PRIMARY KEY column, the column will actually be filled with an integer that is one greater than the largest key already in the table.

No, we don't operate on the ID, but we also don't really have a unique identifier by default. The timestamps, namespace, and dimensions can be freely chosen, there is no restriction. Duplicates would be valid AFAIK.

Still it would be nice to have an internal unique identifier for each metric we insert. Even though we currently don't need it, it may be helpful in the future :)